selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/hotupdate/RecreateUpdate.java

```diff
@@ -0,0 +1,171 @@
+import org.eclipse.che.selenium.core.requestfactory.CheTestAdminHttpJsonRequestFactory;
+import org.eclipse.che.selenium.core.utils.WaitUtils;
+import org.eclipse.che.selenium.core.utils.process.ProcessAgent;
+import org.eclipse.che.selenium.core.workspace.TestWorkspace;
+import org.eclipse.che.selenium.pageobject.CheTerminal;
+import org.eclipse.che.selenium.pageobject.Ide;
+import org.eclipse.che.selenium.pageobject.Menu;
+import org.eclipse.che.selenium.pageobject.ProjectExplorer;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+@Test(groups = {TestGroup.OSIO, TestGroup.MULTIUSER})
+public class RecreateUpdate {
+  private static final String COMMAND_FOR_GETTING_CURRENT_DEPLOYMENT_CHE =
+      "get dc | grep che | awk '{print $2}'";
+
+  @Inject CheTestAdminHttpJsonRequestFactory testUserHttpJsonRequestFactory;
+  @Inject CheTestApiEndpointUrlProvider cheTestApiEndpointUrlProvider;
+  @Inject ProjectExplorer projectExplorer;
+  @Inject OpenShiftCliCommandExecutor openShiftCliCommandExecutor;
+  @Inject private ProcessAgent processAgent;
+  @Inject private TestWorkspace workspace;
+  @Inject private Ide ide;
+  @Inject private SeleniumWebDriver seleniumWebDriver;
+  @Inject private CheTerminal terminal;
+  @Inject private Menu menu;
+
+  private int cheDeploymentBeforeRollout;
+
+  private enum WsMasterStauses {
+    RUNNING,
+    READY_TO_SHUTDOWN,
+    PREPARING_TO_SHUTDOWN,
+    SUSPENDED;
+  }
+
+  @BeforeClass
+  public void setUp() throws IOException {
+
+    cheDeploymentBeforeRollout =
+        parseInt(openShiftCliCommandExecutor.execute(COMMAND_FOR_GETTING_CURRENT_DEPLOYMENT_CHE));
+  }
+
+  @Test
+  public void checkRecreateUpdateStrategy() throws Exception {
+    String ocClientRolloutCommand = "rollout latest che";
+    String restUrlForSuspendingWorkspaces =
+        cheTestApiEndpointUrlProvider.get().toString() + "system/stop";
+    int timeLimitInSecForRecreateingUpdate = 600;
+    int delayBetweenRequest = 6;
+
+    // open a user workspace and send request for preparing to shutdown
+    ide.open(workspace);
+
+    testUserHttpJsonRequestFactory
+        .fromUrl(restUrlForSuspendingWorkspaces)
+        .usePostMethod()
+        .request();
+
+    // make sure, that system is prepared to  shutdown and than ready to shutdown
+    checkExpectedStatusesAndWorkspaceAfterShutDowning();
+
+    // performs rollout
+    openShiftCliCommandExecutor.execute(ocClientRolloutCommand);
+    waitOpenShiftStatus(
+        timeLimitInSecForRecreateingUpdate, delayBetweenRequest, WsMasterStauses.RUNNING);
+
+    // get current version of deployment after rollout
+    int cheDeploymentAfterRollout =
+        parseInt(openShiftCliCommandExecutor.execute(COMMAND_FOR_GETTING_CURRENT_DEPLOYMENT_CHE));
+    // After rollout updating - deployment should be increased on 1. So we previews version +1
+    // should be equal current
+    assertEquals(cheDeploymentAfterRollout, cheDeploymentBeforeRollout += 1);
```

Comment:
I think "+=" it is redundant 
this condition means:  cheDeploymentBeforeRollout = cheDeploymentBeforeRollout + 1 
but in my opinion in this case "cheDeploymentBeforeRollout + 1" is enough.